src/avram/delete_operation.cr

```diff
@@ -0,0 +1,234 @@
+require "./validations"
+require "./callbacks"
+require "./define_attribute"
+require "./operation_errors"
+require "./param_key_override"
+require "./inherit_column_attributes"
+require "./needy_initializer_and_delete_methods"
+
+abstract class Avram::DeleteOperation(T)
+  include Avram::NeedyInitializerAndDeleteMethods
+  include Avram::DefineAttribute
+  include Avram::Validations
+  include Avram::OperationErrors
+  include Avram::ParamKeyOverride
+  include Avram::Callbacks
+  include Avram::InheritColumnAttributes
```

Comment:
My thought was you may want to use those like checking if the user has certain permissions, or maybe checking things like `created_at` to say you can't delete since you just created your account. Dunno, but it could add lots of extra flexibility.